src/backend/distributed/planner/deparse_shard_query.c

```diff
@@ -300,6 +286,16 @@ UpdateRelationsToLocalShardTables(Node *node, List *relationShardList)
 								 relationShardList, QTW_EXAMINE_RTES_BEFORE);
 	}
 
+	if (IsA(node, OnConflictExpr))
+	{
+		OnConflictExpr *onConflict = (OnConflictExpr *) node;
+
+		bool stopWalk = ReplaceRelationConstraintByShardConstraint(relationShardList,
```

Comment:
well, we currently do not have `stopWalk ` concept  at all, so it'd probably simpler just for now
```C
return ReplaceRelationConstraintByShardConstraint(relationShardList,...)
```